backend/src/apiserver/server/run_server.go

```diff
@@ -366,28 +339,16 @@ func (s *RunServer) DeleteRunV1(ctx context.Context, request *apiv1beta1.DeleteR
 
 // Reports run metrics.
 // Applies common logic on v1beta1 and v2beta1 API.
-func (s *RunServer) reportRunMetrics(ctx context.Context, metrics []*model.RunMetric, runId string, experimentId string) ([]map[string]string, error) {
+func (s *RunServer) reportRunMetrics(ctx context.Context, metrics []*model.RunMetric, runId string) ([]map[string]string, error) {
 	err := s.canAccessRun(ctx, runId, &authorizationv1.ResourceAttributes{Verb: common.RbacResourceVerbReportMetrics})
 	if err != nil {
 		return nil, util.Wrap(err, "Failed to authorize the request")
 	}
-	existingRun, err := s.resourceManager.GetRun(runId)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to fetch the requested run")
-	}
-	if experimentId != "" && existingRun.ExperimentId != experimentId && existingRun.ExperimentId != "" {
-		return nil, util.NewInternalServerError(util.NewInvalidInputError("The requested run '%s' belongs to experiment '%s' (requested experiment '%s')", runId, existingRun.ExperimentId, experimentId), "Failed to report run metrics")
-	}
-	if experimentId != "" {
-		if err := s.resourceManager.CheckExperimentBelongsToNamespace(experimentId, existingRun.Namespace); err != nil {
-			return nil, util.Wrap(err, "Failed to report run metrics due to namespace mismatch")
-		}
-		resourceAttributes := &authorizationv1.ResourceAttributes{
-			Namespace: existingRun.Namespace,
-			Verb:      common.RbacResourceVerbList,
-		}
-		if err := s.canAccessRun(ctx, "", resourceAttributes); err != nil {
-			return nil, util.Wrapf(err, "Failed to report run metrics due to authorization error. Check if you have permission to access namespace %s", existingRun.Namespace)
+	// Verify that the run exists for single user mode.
+	// Multi-user model will verify this when checking authorization above.
+	if !common.IsMultiUserMode() {
```

Comment:
Here in `reportRunMetrics()` we verify that the run exists in single-user mode. Should we do the same for delete/archive/unarchive?

Reply:
As the comment suggests, the `canAccessRun()` verifies that for multi-user mode. So we are removing a redundant call to the DB here.